Expanded Postgres.app GDAL instructions on Mac.

vnagendra opened the pull request:
When using Postgres.app and the libraries are still not found (using the default settings currently described in the documentation)

timgraham commented:
I guess this discussion (and even the existing discussion) isn't Django-specific so I wonder if Django's documentation is the best place for such details. Any thoughts about that?

vnagendra commented:
I am not sure why you say it isn't Django specific. I believe it is. PostGIS and GDAL get used in a lot of Django projects. Instructions in Django docs will save users a lot of time and frustration. 

The library loading for GDAL specifically (in settings) is definitely a Django thing.

timgraham commented:
What I mean is that these instructions apply when using non-Django things also. For example, I see a mention of `DYLD_FALLBACK_LIBRARY_PATH` in https://github.com/carsonfarmer/python_geospatial/blob/master/install.md.  I also see a mention of `DYLD_FALLBACK_LIBRARY_PATH` later on in this same document you modified under "Macports." I don't know if things could be better organized so information isn't duplicated.

I'm not a GIS user, but the instructions on this page don't get a lot of updating from what I've seen (e.g. https://code.djangoproject.com/ticket/25633). I'm not opposed to merging this addition, but I'm a bit frustrated at the state of this page in general and so wondering if things could be done better.

vnagendra commented:
Ah very fair point. I am not opposed to looking at the whole page and better organizing it. 

I won't be able to immediately but can definitely do in 2 weeks. 

Being under macports was an error on my part - apologies. It was supposed to be under Postgres.app section. 

If you are good with better organization of the page - air away all the annoyances and I will test and fix them.

timgraham commented on docs/ref/contrib/gis/install/index.txt:
```diff
@@ -240,6 +240,16 @@ the version of PostgreSQL in the Postgres.app you installed:
 You can check if the path is set up correctly by typing ``which psql`` at a
 terminal prompt.
 
+You can also check if the gdal library is installed with your Postgres.app, by
```
No comma

timgraham commented on docs/ref/contrib/gis/install/index.txt:
```diff
@@ -240,6 +240,16 @@ the version of PostgreSQL in the Postgres.app you installed:
 terminal prompt.
 
+You can also check if the gdal library is installed with your Postgres.app, by
+typing `gdal-config --cflags` at the command prompt. 
```
The backticks must be doubled to have an effect on the rendered docs.

timgraham commented on docs/ref/contrib/gis/install/index.txt:
```diff
@@ -240,6 +240,16 @@ the version of PostgreSQL in the Postgres.app you installed:
+typing `gdal-config --cflags` at the command prompt. 
+
+If your GDAL libraries are still not found, you can use `DYLD_FALLBACK_LIBRARY_PATH`, 
+as a fallback by setting it in your `.bashrc` or `.zshrc` etc.,
```
.bash_profile is mentioned earlier -- does this really go in a different file?

timgraham commented on docs/ref/contrib/gis/install/index.txt:
```diff
@@ -240,6 +240,16 @@ the version of PostgreSQL in the Postgres.app you installed:
+
+.. code-block:: bash
+
+    export DYLD_FALLBACK_LIBRARY_PATH=$DYLD_FALLBACK_LIBRARY_PATH:/Applications/Postgres.app/Contents/Versions/9.6/lib
```
X.Y is used earlier rather than 9.6

timgraham commented on docs/ref/contrib/gis/install/index.txt:
```diff
@@ -240,6 +240,16 @@ the version of PostgreSQL in the Postgres.app you installed:
+You can also check if the gdal library is installed with your Postgres.app, by
+typing `gdal-config --cflags` at the command prompt. 
+
+If your GDAL libraries are still not found, you can use `DYLD_FALLBACK_LIBRARY_PATH`, 
```
"still aren't found"
remove comma after DYLD_FALLBACK_LIBRARY_PATH

vnagendra commented:
Yuck, sorry. That's just bad English. No excuse for not checking the rendered doc before submitting PR. Will fix.

vnagendra commented:
@timgraham - I've addressed the comments. Made sure things look right when rendered. I've also added an extra commit to address your comment above re: this page not being updated often. Clearly so, since it's macOS has been renamed to OS X for a few years now. 

If you have any feedback, I am happy to address it. But most of the verbiage is reused from existing parts of the page

timgraham commented:
I thought OS X was renamed to macOS (5e8625ba643db118a44cb32e9e48bf431ef4da53).

vnagendra commented:
Ha, you are right - my bad, will revert that commit entirely.

timgraham commented:
Is this obsolete on newer versions of mac OS? "Apple got rid of LD_LIBRARY_PATH and DYLD_FALLBACK_LIBRARY_PATH." according to http://www.wine-reviews.net/2015/10/winebottler-1752-has-been-released-now.html.

vnagendra commented:
@timgraham  - You are correct. This is no longer relevant AND I mucked up the PR to begin with. I will close it - no action required. Sorry about the bad follow-through.
